### PR TITLE
[MIG] 10.0 mig base location

### PR DESCRIPTION
**Pull request by OpenCode**
Migration of base_location to 10.0

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

File: base_location/__manifest__.py [hunk 1]
```diff
@@ -37,6 +39,7 @@
         'Alejandro Santana <alejandrosantana@anubia.es>',
         'Sandy Carter <sandy.carter@savoirfairelinux.com>',
         'Yannick Vaucher <yannick.vaucher@camptocamp.com>',
+        'Francesco Apruzzese <f.apruzzese@apuliasoftware.it>',
```
Comment (andrea4ever) [inline base_location/__manifest__.py]: Please move contributor on README

**Review by andrea4ever — APPROVED** [review]

File: base_location/__manifest__.py (hunk 1 above)
Comment (OpenCode) [reply base_location/__manifest__.py]: Add contributor in README

**Review by eLBati — APPROVED** [review]

**Comment by pedrobaeza** [comment]
We should wait until https://github.com/odoo/odoo/pull/13835 is resolved, as the module in that PR overlaps most of the features of this module.

**Comment by alexis-via** [comment]
We need to decide RAPIDLY what we do about the dependency on "sales_team":
1) should we add a dependency on sales_team just to be able to add a menu entry in Sales > Configuration > Address Book ?
2) or should we stay with a dependency on "base" only and put the menu entry elsewhere, for example under base.menu_custom (i.e. "Settings > Technical > xxx").
3) any other bright idea ?
We face this problem in many module, so we need to find the right solution now.

**Review by alexis-via — APPROVED** [review]
Code review, no test. Approve with my remark about the dependency on sales_team ; I would prefer to avoid it !

**Comment by pedrobaeza** [comment]
One suggestion is to copy the XML menu definition with the same XML-IDs here, so that we don't overlap when sale_team in installed.

**Comment by alexis-via** [comment]
@OpenCode Did you migrate base_location_geonames_import too ? I don't see any PR that migrate this module, but I see your PR to migrate l10n_it_base_location_geonames_import... Otherwise, I'll do a PR to port base_location_geonames_import today or tomorrow.

**Comment by OpenCode** [comment]
@alexis-via I need sale_team to use Sales > Configuration > Address Book menu. I don't like this dependence too, my I need it.
I will port base_location_geonames_import after this pull request but if you can do it for me it's ok :)

**Comment by alexis-via** [comment]
@OpenCode Yes, but @pedrobaeza suggests a solution to work around this problem, even if I'm not sure that it really works. @pedrobaeza did you already test the solution that you suggest ?

**Comment by alexis-via** [comment]
@pedrobaeza Could you make a PR against this PR to show us exactly the solution you propose ? I tried to implement it, but it's very difficult in real life:
- it's not just 1 menu entry, but all the "sales" hierarchy: sales_team.menu_base_partner > sales_team.menu_sale_config > sales_team.menu_config_address_book > sales_team.menu_localisation
- these menu entries use groups such as group_sale_salesman or group_sale_manager, which are also part of the sales_team module
  So I fear that it's more complex than what you think when you try to implement it.

**Comment by alexis-via** [comment]
@OpenCode I ported base_location_geonames_import in this PR https://github.com/OCA/partner-contact/pull/320

**Comment by OpenCode** [comment]
@alexis-via Thanks :)

@alexis-via @pedrobaeza I don't like to use this workaround

**Comment by pedrobaeza** [comment]
OK, the workaround implies too much records to replicate.

**Comment by alexis-via** [comment]
@pedrobaeza OK, which option do we take ? I proposed 3 options in an earlier comment... This impacts a lot of OCA modules, so we'd better choose the right option now.

**Comment by pedrobaeza** [comment]
As base module doesn't have anything except Settings menu, we should locate there the corresponding menu if we don't want to depend in any other module, but maybe people with no access to Settings should be able to modify zip entries. A new top menu is no better solution, as is going to be underused. The less bad is then the Settings one I think.

**Comment by OpenCode** [comment]
I propose to keep sale_team as dependence. We can ask to Odoo (using a PR) to move the menu in the next version.

**Comment by eLBati** [comment]
Depending on `sales_team` sounds strange to me too, however `Countries` and other localization menus are added by `sales_team`, so, if we follow how odoo 10 is designed, adding `Cities` menu under `sales_team.menu_localisation` should be fine.
As a user, I would look for `Cities` in the neighbourhood of `Countries`, so, having to go to `Settings` would be strange

**Comment by alexis-via** [comment]
@eLBati We all know that there is not perfect solution:
- if we prefer a well-designed configuration menu, we choose option 1
- if we prefer well-designed dependencies, we choose option 2
  I tend to prefer option 2, but if OCA opts for option 1, it's not a big deal for me.

If someone has a bright idea for an option 3, it's time to post it here now !

**Comment by andrea4ever** [comment]
I agree with @alexis-via to use option 1

**Comment by pedrobaeza** [comment]
@alexis-via, any progress in this?

**Comment by alexis-via** [comment]
@pedrobaeza What do you expect me to do ? I'm not the author of this PR, and it don't think any decision has been made between the 2 options (depend on sales_team or not). But it would be good to take a decision and move on.

I said in my previous comment that I prefered option 2. @andrea4ever said he preferd option 1. Here we are...

**Review by lasley — COMMENTED** [review]
I'm for option 2, moving to Settings => Technical or somewhere. This menu doesn't need to be seen by a normal user, and adding a depends just for a technical menu seems a bit overboard IMO

**Comment by rafaelbn** [comment]
Hi,

@andrea4ever said in https://github.com/OCA/partner-contact/pull/317#issuecomment-256022714 that "agree with Alexis option 1"
@alexis-via said in https://github.com/OCA/partner-contact/pull/317#issuecomment-264030902 that "I prefered option 2"

Something doesn't fit 😉 

My opinion is option 2 -> Settings

**Comment by pedrobaeza** [comment]
@OpenCode, can you do then the honors to put the menu under Settings?

**Comment by OpenCode** [comment]
Ok

ok ok ok

**Review by alexis-via — COMMENTED** [review]
To avoid this warning "Field res.company.better_zip_id: parameter 'select' is no longer supported; use 'index' instead." you should remove select=1 in company.py (or convert it to index=True, but I don't think it's useful here)

**Comment by rafaelbn** [comment]
Thanks @OpenCode for this migration, waiting for last changes 😄

**Comment by eLBati** [comment]
@pedrobaeza @OpenCode @alexis-via @rafaelbn @lasley , moved menu under settings

**Comment by alexis-via** [comment]
OK, let's merge then :+1:

**Comment by pedrobaeza** [comment]
Please, let me squash

**Pull request merged**